### PR TITLE
refactor(common): time_constraints

### PR DESCRIPTION
**Pull request by alghanor**
unify across project

https://tillhub.atlassian.net/browse/IOS-2945

### CODE REVIEW

File: lib/v1/products/prices/book/constraints/index.js [hunk 1]
```diff
@@ -1,153 +1,18 @@
...
-}
+const { oneOf } = require('../../../../../helpers/payload-or-null')
+const { timeConstraints } = require('../../../../../common/time_constraints')
 
 module.exports = {
   type: 'object',
   additionalProperties: false,
   properties: {
-    time: {
+    time: oneOf({
```
Comment (mohsen89z) [inline lib/v1/products/prices/book/constraints/index.js:8]: Why this one has a `oneOf`?

**Review by mohsen89z — APPROVED** [review]

File: lib/v1/products/prices/book/constraints/index.js (hunk 1 above)
Comment (alghanor) [reply lib/v1/products/prices/book/constraints/index.js:8]: We regard most things optional these days, would you like to enforce existence?

Comment (mohsen89z) [reply lib/v1/products/prices/book/constraints/index.js:8]: It's backward compatible, so it's fine :D

**Pull request merged**